src/CloudNative.CloudEvents/DelegatingFormatter.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright 2025 Cloud Native Foundation.
+// Licensed under the Apache 2.0 license.
+// See LICENSE file in the project root for full license information.
+
+using CloudNative.CloudEvents.Core;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Mime;
+using System.Threading.Tasks;
+
+namespace CloudNative.CloudEvents;
+
+// TODO:
+// - Naming:
+//   - Is the namespace appropriate? We have very few types in this root namespace
+//   - Delegating or Decorating?
```

Comment:
I'm torn between the two but I think `Delegating` is the right way to go here. Decorating implies that data is modified/enhanced in some way whereas delegating does a better job of describing it's strictly behavioral, IMO.